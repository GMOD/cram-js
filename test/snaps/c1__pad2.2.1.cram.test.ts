import { testFile } from '../testUtil'
testFile('c1#pad2.2.1.cram')
