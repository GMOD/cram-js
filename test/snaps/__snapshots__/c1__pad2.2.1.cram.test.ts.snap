// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#pad2.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10098,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10098,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              4,
            ],
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "DL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              2,
              2,
              2,
              2,
            ],
            "numCodes": 4,
            "numLengths": 4,
            "symbols": [
              1,
              2,
              3,
              5,
            ],
          },
          "parametersBytes": 10,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              80,
              68,
              73,
            ],
          },
          "parametersBytes": 8,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              2,
              2,
              2,
              3,
              3,
            ],
            "numCodes": 5,
            "numLengths": 5,
            "symbols": [
              0,
              3,
              5,
              1,
              2,
            ],
          },
          "parametersBytes": 12,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              3,
              3,
              4,
              4,
              4,
              4,
            ],
            "numCodes": 7,
            "numLengths": 7,
            "symbols": [
              0,
              2,
              6,
              1,
              3,
              4,
              5,
            ],
          },
          "parametersBytes": 16,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "PD": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              2,
              1,
              3,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              3,
              3,
              4,
              4,
              4,
              4,
            ],
            "numCodes": 7,
            "numLengths": 7,
            "symbols": [
              10,
              13,
              16,
              2,
              3,
              8,
              14,
            ],
          },
          "parametersBytes": 16,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10154,
      "_size": 15,
      "alignmentSpan": 10,
      "landmarks": [
        246,
      ],
      "length": 474,
      "numBases": 129,
      "numBlocks": 6,
      "numLandmarks": 1,
      "numRecords": 13,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 10154,
        "_size": 15,
        "alignmentSpan": 10,
        "landmarks": [
          246,
        ],
        "length": 474,
        "numBases": 129,
        "numBlocks": 6,
        "numLandmarks": 1,
        "numRecords": 13,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0a",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10406,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0b",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10407,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0c",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10408,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0d",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10409,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "GTTAAC",
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 16,
            "readName": "s1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10410,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "P",
                "data": 1,
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TTAA",
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 1,
                "pos": 10,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 14,
            "readName": "s2",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10411,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "GTT",
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 3,
                "pos": 9,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 13,
            "readName": "s3",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10412,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "P",
                "data": 3,
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "AAC",
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 13,
            "readName": "s4",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10413,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 1,
                "pos": 5,
                "refPos": 5,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TA",
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 7,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 7,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s5",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10414,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 3,
                "pos": 3,
                "refPos": 3,
              },
              {
                "code": "I",
                "data": "GTTAAC",
                "pos": 3,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 3,
                "pos": 9,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s6",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10415,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 2,
                "pos": 5,
                "refPos": 5,
              },
            ],
            "readGroupId": -1,
            "readLength": 8,
            "readName": "s7",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10416,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 5,
                "pos": 1,
                "refPos": 1,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TA",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 3,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 5,
                "pos": 3,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 2,
            "readName": "s8",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10417,
          },
          {
            "alignmentStart": 5,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 1,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "P",
                "data": 2,
                "pos": 2,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TA",
                "pos": 2,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 4,
                "refPos": 6,
              },
            ],
            "readGroupId": -1,
            "readLength": 3,
            "readName": "s9",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 10418,
          },
        ],
        "header": {
          "_endPosition": 10436,
          "_size": 31,
          "compressedSize": 31,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            10,
            13,
            0,
            4,
            3,
            11,
            12,
            13,
            255,
            255,
            255,
            255,
            15,
            226,
            36,
            189,
            138,
            112,
            229,
            70,
            108,
            106,
            3,
            59,
            232,
            134,
            4,
            92,
            133,
          ],
          "contentId": 0,
          "contentPosition": 10405,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10185,
            "_size": 31,
            "contentIds": [
              11,
              12,
              13,
            ],
            "md5": [
              226,
              36,
              189,
              138,
              112,
              229,
              70,
              108,
              106,
              3,
              59,
              232,
              134,
              4,
              92,
              133,
            ],
            "numBlocks": 4,
            "numContentIds": 3,
            "numRecords": 13,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 10,
            "refSeqStart": 1,
          },
          "uncompressedSize": 31,
        },
      },
    ],
  },
]
`;
